src/components/pool/PoolPageHeader.vue

```diff
@@ -124,6 +124,19 @@ const poolTypeLabel = computed(() => {
 
   return key ? t(key) : t('unknownPoolType');
 });
+
+const nonAllowedRateProviders = computed(() => {
+  return (
+    props.pool?.poolType === 'Weighted' &&
+    props.pool?.priceRateProviders?.some(
+      provider =>
+        !ALLOWED_PRICE_RATE_PROVIDERS['*'][provider.address] &&
+        !ALLOWED_PRICE_RATE_PROVIDERS[provider.token?.address]?.[
+          provider.address
+        ]
+    )
```

Comment:
Personally I find this easier to parse/understand, but it might be worth getting other people's opinions
```suggestion
    !props.pool?.priceRateProviders?.every(
      provider =>
        ALLOWED_PRICE_RATE_PROVIDERS['*'][provider.address] ||
        ALLOWED_PRICE_RATE_PROVIDERS[provider.token?.address]?.[
          provider.address
        ]
    )
```